src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala

```diff
@@ -176,30 +179,21 @@ class ClusterMonitorActor(val cluster: Cluster,
     * @return ShutdownActor
     */
   private def handleFailedCluster(errorDetails: ClusterErrorDetails, instances: Set[Instance]): Future[ClusterMonitorMessage] = {
-    val deleteFuture = Future.sequence(Seq(
-      // Delete the cluster in Google
-      gdDAO.deleteCluster(cluster.googleProject, cluster.clusterName),
-      // Remove the service account key in Google, if present.
-      // Only happens if the cluster was NOT created with the pet service account.
-      removeServiceAccountKey,
-      // create or update instances in the DB
-      persistInstances(instances),
-      // save cluster errors to the DB
-      dbRef.inTransaction { dataAccess =>
-        val clusterId = dataAccess.clusterQuery.getIdByUniqueKey(cluster)
-        clusterId flatMap {
-          case Some(a) => dataAccess.clusterErrorQuery.save(a, ClusterError(errorDetails.message.getOrElse("Error not available"), errorDetails.code, Instant.now))
-          case None => {
-            logger.warn(s"Could not find Id for Cluster ${cluster.projectNameString}  with google cluster ID ${cluster.dataprocInfo.googleId}.")
-            DBIOAction.successful(0)
-          }
-        }
-      }
-    ))
+    for {
+      _ <- Future.sequence(Seq(
```

Comment:
minor here, we should almost never use `Seq`, it's mutable. `List` will be better